### ISSUE TITLE
Stage/checkpoint, mainly for start-point specification

### ISSUE THREAD

**Issue by vreuter** [databio/dnameth_pipelines#44]
Cleaning up files motivates this. For example, right now removal of unzipped trimmed FASTQ files will fool the pipeline into thinking that trimming's not been completed, but it'd be nice to be able to simply tell the pipeline to start at a point beyond that.

**Comment by vreuter** [comment]
The specific example noted here, of raw FASTQ removal and read trimming, is especially well situated to benefit from this, as the `fastq_folder` usage, at least in `wgbs.py`, is bounded at the alignment phase, i.e. is used strictly before alignment.

**Comment by nsheff** [comment]
yep! this is clearly an issue and perfect use case. So far I've shied away from directly addressing it because it's a slippery slope down to a full-blown professional pipeline system, but if we had a very simple solution that could partially solve the problem I'd be for it. this kind of stuff can get really complicated really quickly.

**Comment by nsheff** [comment]
for some reason I thought this was on the pypiper repository. I think I misinterpreted your comment, sorry.

**Comment by vreuter** [comment]
Well the capabilities do live in `pypiper`, but I opened this to suggest that we harness them here.
